### ISSUE TITLE
Can this be used for COVID-19 tracing?

### ISSUE THREAD

**Issue by zukunft** [owntracks/android#807]
Would it be possible to use this app for anonymous COVID-19 tracing?

The setup could be
1. the handy app creates an unique random id number and uses it with a secure connection for anonymous login on the owntracks server. Only the handy app knows the id number, so that no trace back from the server to the handy is not possible. So if the handy app is deleted the connection is lost. This would allow all users to have full control over their data.
2. the handy app sends the position data with the id number to the owntracks server
3. the owntracks server saves the positions data for a predefined period e.g. 14 days and automatically deletes all position data older than 14 days
4. A flag is added to the handy app, where the user can say "I have been tested positive for COVID-19". Once confirmed by the user, this status information is send to the owntracks server with the id number.
5. If the owntracks server receives a status change of an id number it creates a "list of potential COVID-19 positive id numbers" based on some settings e.g. at least 15 minutes within 2 meters of the id number with the new positive status
6. If the handy app sends new position data, it also checks if the id number has been added to the "list of potential COVID-19 positive id numbers" and if yes, show a "testing invitation" to the handy user.

I have not yet checked, which parts are already implemented and which parts needs to be added. Any held is appreciated!

**Comment by jpmens** [comment]
I believe this would be possible, and I think most of the bits are in place. What would be needed is instead of user/device we currently use maybe a generated UUID for the anonymity. What would also be needed is what you call the "owntracks server"; this is something which currently doesn't exist because we don't have a central service. Of course for a situation such as the one you describe this would be possible.

Likely the more work-intensive bit is adding the "I've tested positive" flag to the UI.

**Comment by zukunft** [comment]
Thanks for the answer. What would be the next steps: Add the "I've tested positive" to the GUI? If yes, should I open an issue?
Maybe this could be coordinated with https://www.pepp-pt.org/ . I hope, that they open a public repository soon.

**Comment by jpmens** [comment]
Are you expecting us to do the work for you? I don't think that's how this works. At best you can begin by explaining *why* we would want to do that.

**Comment by zukunft** [comment]
"you" is just me personal. I am not related to pepp-pt. I only reason why I am asking this is because I want to be free to leave my house soon.
I have read this article https://smw.ch/article/doi/smw.2020.20225 and it looks to me, that a tracing app is missing until now. I have been using owntracks for myself in the past and my idea is, that this could be reused for this purpose. 
Maybe pepp-pt will solve the problem, but at the moment they are a kind off closed, so I don't know, if they will have something useful soon. And as you said, owntracks seems to be a good base to start. So the why is very simple and the target is not that you work for me or anybody else. The target is that we work together.

**Comment by jpmens** [comment]
I hope for the privacy of all of us, that no European government enforces contact tracing/tracking, as it's going to open a snake-pit of privacy issues. This cannot be done anonymously: think of leaving/entering your house two or three times and having your data in a central location, then you are pinpointable.

We created OwnTracks for privacy; it's in the name "my own tracks". I for one am not interested in being part of this.

Closing this issue.
